website/guides/sgb_border.md

```diff
@@ -0,0 +1,178 @@
+- Transfer and display the border's tilemap and palettes
+
+<small>
+
+Written by [sylvie (zlago)](https://zlago.github.io/me/), idea (and minor help) by [valentina (coffee bat)](https://coffeebat.neocities.org/), reviews and improvements by [ISSOtm](https://eldred.fr) and [avivace](https://github.com/avivace).
+
+</small>
+
+## Enabling SGB features
+
+Before we can do anything else, we must first specify in the header that this game is aware of SGB features.
+Otherwise, the SGB BIOS will ignore any packets we send, and we won't even be able to detect when the program is running on SGB.
+
+To enable SGB features:
+
+- The SGB flag (`$0146`) must be set to `$03`;
+- The old licensee code (`$014b`) must be set to `$33`.
+
+This can be achieved by passing the `--sgb-compatible` and `--old-licensee 0x33` flags to [`rgbfix`](https://rgbds.gbdev.io/docs/rgbgfx.1).
+
+## Packets
+
+The SGB BIOS can be "talked to" via *command packets*, sent bit by bit via the `P1`/`JOYP` register.
+
+An SGB packet consists of:
+
+- a "reset pulse" (`P1`=`%xx00xxxx`)
+- 128 data pulses (`P1`=`%xx01xxxx` for "1", `P1`=`%xx10xxxx` for "0")
+- a "0" pulse (`P1`=`%xx10xxxx`)
+
+You must set `P1` to `%xx11xxxx` between each pulse.
+
+This adds up to 16 Bytes of data (LSB first), If a packet doesn't read all 16 bytes, the unused bytes are ignored.
+
+**You should wait 4 frames between each packet.**
+
+For an example of such routine, see this [code](https://github.com/zlago/violence-gbc/blob/11cfdb6ee8a35e042fa9712484d814e0961cea7c/src/sub.sm83#L413-L463) and the related Pan Docs entry: [SGB Command Packet on Pandocs](https://gbdev.io/pandocs/SGB_Command_Packet.html).
+
+This guide glosses over a minor detail, as certain packets can be (albeit unccomon) more than 16 bytes.
+
+## TRN
+
+Bulk transfer (TRN) packets tell the SGB to copy the contents of the screen to buffers in Super NES work RAM. The `CHR_TRN` and `PCT_TRN` packets are used to send data for SGB borders.
+
+For that to function properly, you must:
+
+1. set `BGP` to `$e4` and `LCDC` to `$91` (screen enabled, BG uses tiles `$8000`-`$8fff` and tilemap `$9800`, WIN and OBJ disabled, BG enabled)
+2. set `SCX` and `SCY` to `$00`
+3. the tilemap consists of `$00`, `$01`..`$13`, 12 bytes padding (offscreen), `$14`..`$27`, padding, repeat until `$ff` (inclusive)
+4. the data you want to send must be loaded at `$8000`-`$8fff`
+
+You can do 1, 2 and 3 via [this snippet](https://github.com/zlago/snek-gbc/blob/baef0369f57d2b0d58316cb1c28c6cc22475a6c9/code/init.sm83#L208-L230)
+
+- **You must load the data into VRAM and enable the screen before sending the TRN packet**
+- **You must wait ~8 frames after each TRN** instead of just 4
+
+## Detecting SGB
+
+Here's how a SGB detection routine should look like:
+
+- wait 12 or more frames after the console boots
+- send a `MLT_REQ`, selecting 2 or 4 players (`$89, $01` or `$89, $03`), and wait at least 1 frame
+- attempt to advance the read player (set `P1.5` to 0 then 1)
```

Comment:
This sounds confusing. What does it mean?

Reply:
My guess:
1. wait for the SGB to finish initializing
2. set multiplayer mode and wait for the SGB to recognize the request to go to multiplayer mode
3. test whether multiplayer mode was activated by reading the controller a couple times and seeing whether the input register reflects a player other than player 1
4. optionally turn off multiplayer mode

Reply:
@ISSOtm